### PR TITLE
feat - Support adding children on the right side of Alerts

### PR DESCRIPTION
**Pull request by ccedrone**
Issue: # [992](https://github.com/cengage/react-magma/issues/992)

## What I did
Added `additionalContent` prop to Alerts and Banners for right aligned children elements such as buttons and badges.

## Screenshots (if appropriate):
example

## Checklist 
- [x] changeset has been added
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/react-magma-dom/src/components/AlertBase/index.tsx [hunk 1]
```diff
@@ -275,16 +276,39 @@ const StyledAlertInner = styled.div<AlertBaseProps>`
...
 
+export const AdditionalContentWrapper = styled.div`
+  flex: 1;
+  justify-content: flex-end;
+  display: flex;
+  * {
+    display: table-cell;
+    margin: 0;
+    margin-left: ${props => props.theme.spaceScale.spacing03};
+  }
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/AlertBase/index.tsx:309]: Why is this needed? It seems like I can remove these styles and it still works the same

File: packages/react-magma-dom/src/components/AlertBase/index.tsx [hunk 2]
```diff
@@ -444,6 +471,7 @@ export const AlertBase = React.forwardRef<HTMLDivElement, AlertBaseProps>(
         theme={theme}
         variant={variant}
       >
+        {console.log(additionalContent)}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/AlertBase/index.tsx:474]: We should remove this

File: packages/react-magma-dom/src/components/Banner/index.tsx [hunk 3]
```diff
@@ -187,6 +194,7 @@ export const Banner = React.forwardRef<HTMLDivElement, BannerProps>(
     const {
       actionButtonText,
       actionButtonOnClick,
+      additionalContent,
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Banner/index.tsx:197]: This needs to be added to the BannerProps

**Review by silvalaura — COMMENTED** [review]
We should also add some unit tests and examples in the docs site

File: packages/react-magma-dom/src/components/AlertBase/index.tsx (hunk 1 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/AlertBase/index.tsx:309]: If you look at the Storybook example with two badges, and also the button, it cleans up the left spacing and fixes an issue with left button padding.

Comment (ccedrone) [reply packages/react-magma-dom/src/components/AlertBase/index.tsx:309]: Fair enough, I'll remove that then!

**Review by silvalaura — APPROVED** [review]

**Review by orion-cengage — CHANGES_REQUESTED** [review]
Couple things I noticed.

- Can we make sure there is 16px of space/margin/whatever to the left of the container with the badge or button, so the text never runs into the item?
- For some reason, the badges are wrapping and that's not supposed to happen. They should stay a single line like the button. Is this a problem with the alert, or the badge component?

image

**Comment by ccedrone** [comment]
Flexbox needed a tweak, added a left margin of 12px to the additional children container as well.

**Review by orion-cengage — APPROVED** [review]

**Review by katieweinstein — APPROVED** [review]

**Pull request merged**